### DISCUSSION TITLE
Hybrid Mobile Frameworks

### DISCUSSION THREAD

**Discussion by luisaveiro** [luisaveiro/playground#5] [Ideas]
The Playground repository includes over 15 tech stacks which are primarily web-based. I want to add Hybrid Mobile Frameworks to the repository. To reduce the set-up complexity that some Hybrid Mobile Frameworks require to create a new project. 

I have identified the following frameworks that I would like to try adding to Playground:

- [ ] Flutter
- [x] React Native
- [x] Ionic
- [ ] NativeScript

I have minimal experience working with the suggested Hybrid Mobile Frameworks. I would like feedback on known set-up challenges or limitations. I'm aware of developing hybrid mobile apps in a Dockerised environment is not common practice.

**Comment by luisaveiro** [comment]
I was able to get [React Native](https://github.com/luisaveiro/playground/tree/main/samples/react-native) working with [Expo Go](https://expo.dev/client) - iOS Client.

The Expo dev CLI automatically sets the dev server IP address to the Docker container, which prevents the Expo Go client from accessing the project. To resolve this issue, you need to force the URL to be your host IP address with the `EXPO_PACKAGER_PROXY_URL` environment variable.

I have made the `EXPO_PACKAGER_PROXY_URL` environment variable configurable within the Docker Compose file using a DotEnv file.

**compose.yml**
```yaml
services:
  # The React Native app container.
  react-native:
    environment:
      EXPO_PACKAGER_PROXY_URL: http://${PROXY_URL}:${APP_PORT:-80}
```

**DotEnv**
```ini
#--------------------------------------------------------------------------
# Network env
#--------------------------------------------------------------------------

# Map the app container exposed port to the host port. | default: 80
APP_PORT=80

# Map the Expo dev server URL to the host IP Address. | e.g. 192.168.1.100
PROXY_URL="192.168.1.100"
```

**Reply by geoman2** [reply]
Hi Luis, did you try to compile to get compiled ios .ipa files?

**Reply by luisaveiro** [reply]
@geoman2 I have not tried compiling ios .ipa files. The purpose of this project is to provide a starting point for developers. I have not explored further (production-ready) configuration.
